charts/argus/templates/_helpers.tpl

```diff
@@ -130,4 +130,67 @@ Collector Pod security context
 capabilities:
   drop: {{ toYaml .Values.collector.securityContext.capabilities.drop | nindent 4 }}
   add: {{ toYaml $addCaps | nindent 4 }}
-{{- end }}
\ No newline at end of file
+{{- end }}
+
+{{/*
+LM Credentials and Proxy Details
+*/}}
+
+{{- define "lm-credentials-and-proxy-details" -}}
+{{- $secretObj := (lookup "v1" "Secret" .Release.Namespace .Values.global.userDefinedSecret) | default dict }}
+{{- $secretData := (get $secretObj "data") | default dict }}
+- name: ACCESS_ID
+  valueFrom:
+    secretKeyRef:
+      name: {{ include "argus-secret-name" . }}
+      key: accessID
+- name: ACCESS_KEY
+  valueFrom:
+    secretKeyRef:
+      name: {{ include "argus-secret-name" . }}
+      key: accessKey
+- name: ACCOUNT
+  valueFrom:
+    secretKeyRef:
+      name: {{ include "argus-secret-name" . }}
+      key: account
+- name: ETCD_DISCOVERY_TOKEN
+  valueFrom:
+    secretKeyRef:
+      name: {{ include "argus-secret-name" . }}
+      key: etcdDiscoveryToken
+{{- if or $secretData.argusProxyUser $secretData.proxyUser .Values.proxy.user }}
+- name: PROXY_USER
+  valueFrom:
+    secretKeyRef:
+      name: {{ include "argus-secret-name" . }}
+      {{- if $secretData.argusProxyUser }}
+      key: argusProxyUser
+      {{- else }}
+      key: proxyUser
+      {{- end }}
+{{- end }}
+{{- if or $secretData.argusProxyPass $secretData.proxyPass .Values.proxy.pass }}
+- name: PROXY_PASS
+  valueFrom:
+    secretKeyRef:
+      name: {{ include "argus-secret-name" . }}
+      {{- if $secretData.argusProxyPass }}
+      key: argusProxyPass
+      {{- else }}
+      key: proxyPass
+      {{- end }}
+{{- end }}
+{{- end }}
+
+
+{{/*
+Return secret name to be used based on the userDefinedSecret.
+*/}}
+{{- define "argus-secret-name" -}}
```

Comment:
Can we move this to `lm-util` by renaming it to something like `lmutil.secret-name`?

Template refers the global section, nothing specific so good to have this there, so one secret to configure all child charts (looking out from usability perspective)

Reply:
Make sense. Noted